### PR TITLE
Fix predicate evaluation in OutputCacheAttribute

### LINKED ISSUE #44739: .NET Core 7 Output Cache does not work with app.UseEndpoints() properly
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When configuring the endpoints using **app.Map** everything works as expected. But when the endpoints belong to a controller that is configured with **app.UseEndpoints()** the cache does not ignore one of  the paths to the controller.

I am using a call to UseEndpoints to configure a controller with two methods, both of which have the OutputCache attribute set to a preconfigured policy based on a dynamic condition (in this case it just checks the URL. My real case scenario is somewhat different and checks for a flag in the HttpContext.Items collection). 

Here are the tested URLS  (test app is attached):

// expected to be cached, works, uses **app.Map**
/TimeCached

// expected to be not cached, works, uses **app.Map**
/TimeNotCached

// expected to be cached, works, uses **app.UseEndpoints()**
/Home/TimeCached

// expected not to be cached, **does not work** , uses **app.UseEndpoints()**
/Home/TimeNotCached

### Expected Behavior

I would expect that the call to app.UseEndpoints() would behave the same way as app.Map() and respectively **cache** and **not cache** depending on a dynamic condition within the configured policy.

### Steps To Reproduce

Just run the code sample and browse the following URLS:

// expected to be cached, works, uses **app.Map**
/TimeCached

// expected to be not cached, works, uses **app.Map**
/TimeNotCached

// expected to be cached, works, uses **app.UseEndpoints()**
/Home/TimeCached

// expected not to be cached, **does not work** , uses **app.UseEndpoints()**
/Home/TimeNotCached

### Exceptions (if any)

_No response_

### .NET Version

7.0.100-rc.2.22477.23

### Anything else?

[output-cache-tests.zip](https://github.com/dotnet/aspnetcore/files/9868168/output-cache-tests.zip)

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #44739

When using a named policy from a controller/action, it currently injects a `DefaultPolicy` before it, which then won't be part of the predicate if there is one specified. The `DefaultPolicy` should only be included if a named policy is not defined, or if the named policy is using one internally (by default).

### CODE REVIEW

**Review by adityamandaleeka — APPROVED** [review]

File: src/Middleware/OutputCaching/test/OutputCacheAttributeTests.cs [hunk 1]
```diff
@@ -57,6 +57,20 @@ public async Task Attribute_CreatesNamedPolicy()
         Assert.Equal(42, context.ResponseExpirationTimeSpan?.TotalSeconds);
     }
 
+    [Fact]
+    public async Task Attribute_NamedPolicyDoesntInjectDefaultPolicy()
```
Comment (BrennanConroy) [inline src/Middleware/OutputCaching/test/OutputCacheAttributeTests.cs:61]: super nit:
```suggestion
    public async Task Attribute_NamedPolicyDoesNotInjectDefaultPolicy()
```

**Review by BrennanConroy — APPROVED** [review]
Yay! It wasn't a routing issue 😄

**Pull request merged**